Generate documentation file for config group

machi1990 opened the pull request:
**1st commit:**
Make it possible to generate a file par config group. 

**2nd commit:** 
use the generated file for `FormAuthConfig` to avoid the usage of an inexistant file which  was generating the following warning
```
asciidoctor: INFO: security-guide.adoc: line 69: optional include dropped because include file not found: target/asciidoc/generated/config/quarkus-http-auth-form.adoc 
```

**3rd commit:**
Resolves https://github.com/quarkusio/quarkus/issues/5097

machi1990 commented on docs/src/main/asciidoc/security-guide.adoc:
```diff
@@ -64,10 +64,6 @@ value. This cookie contains a expiry time as part of the encrypted value, so all
 clocks synchronised. At one minute intervals a new cookie will be generated with an updated expiry time if the session
 is in use.
 
-The following properties can be used to configure form based auth:
```
Should we add a link to the form-auth section of all-doc-page?

gsmet replied:
These properties are not generated in a smaller file?

machi1990 replied:
IIRC, they are not generated in a smaller file.

gsmet replied:
Hmm, I don't understand. I remember having all of them in small files. Is it gone?

gsmet replied:
Would it be possible to generate a file per config group too? Looks like we have quite a lot of docs where it makes sense.

machi1990 replied:
+1, I was leaning toward that path. I'll look at it.

FroMage replied:
Why is it inexistant? I reviewed that guide last week and didn't see any inexistant file?

machi1990 replied:
The `FormAuthConfig` configuration is a config group, and we do not generate specific doc files for config groups. If it was previously there, maybe the file being referenced here was for a config root that is no longer there.

FroMage replied:
Ah, that's possible.

machi1990 committed:
docs: generate a specific documentation file for each config group

The file can be included via \include::{generated-dir}/config/quarkus-your-extension-dashed-config-group-simple-name.adoc[opts=optional, leveloffset=+1]

machi1990 committed:
docs: use FormAuthConfig generated file instead of an inexistant file

machi1990 commented:
@gsmet I have added a commit which enable the generation of config group as discussed above.

machi1990 committed:
docs: avoid generating partial documentation

gsmet reviewed (approved):
I checked the output and everything looks in order. We can tune it later if needed.
